### DISCUSSION TITLE
Help with regex match

### DISCUSSION THREAD

**Discussion by AkikoOrenji** [synfron/ReshaperForBurp#29] [General]
I'm trying to use Set Variable to extract a token from a stored server response (manually generated with Send Request and Capture Output to global variable) and place it into another variable i can use in other requests (or directly using Set Value on those request).

I notice the Regex Pattern will allow me to select a pattern from the source text e.g. \_token"\:"(.*?)"\,"token\_ (grab everything between the \_token and token\_)  and replace it with a Regex Replacement Text but what i really need is to save it into a new variable. Leaving the Regex Replacement Text box blank results in the new variable including everything from the original text minus the value i actually wanted and what matched with the regex. 

Any ideas on a better approach for grabbing the data i need out of a Text Value into a variable ? 

Thanks

**Comment by ddwightx** [comment]
Hi @AkikoOrenji,
You could change your regex capture groups to instead capture the tokens that you want rather than the text in between.

For example, this regex `([a-z]+)\d+([a-z]+)` matches letters, followed by numbers, followed by letters. The parentheses are around the letter patterns rather than the number pattern, so my groups will capture the letters. Consider the Source Text `test123string`. If I set the Regex Replacement Text as `$1 $2` the variable will have the text `test string`.

That being said, If your source text is JSON, depending on what exactly you need to extract, you don't need to use the regex replace functionality, but instead can set JSON as the Source Value Type and then use a JSON Path as the Source Value Path to extract the value from the JSON that you want. For example, if I have the JSON `{ "somekey": "someValue", "token": "sometoken"}`, the Source Value Path `$.token` will extract the value `sometoken`.

**Reply by AkikoOrenji** [reply]
Thanks for this. More elegant by far.
